apps/bfd-server/bfd-server-war/src/main/java/gov/cms/bfd/server/war/r4/providers/pac/PhaseResourceFilter.java

```diff
@@ -0,0 +1,35 @@
+package gov.cms.bfd.server.war.r4.providers.pac;
+
+import gov.cms.bfd.server.war.r4.providers.pac.common.ResourceFilter;
+import gov.cms.bfd.server.war.r4.providers.pac.common.ResourceTypeV2;
+import gov.cms.bfd.sharedutils.config.SemanticVersion;
+import gov.cms.bfd.sharedutils.config.SemanticVersionRange;
+import lombok.AllArgsConstructor;
+
+/**
+ * Restricts results to only claims having phase versions within a specific {@link
+ * SemanticVersionRange}.
+ */
+@AllArgsConstructor
+public class PhaseResourceFilter implements ResourceFilter {
+  /** The range of semantic version numbers that are allowed to be returned to clients. */
+  private final SemanticVersionRange allowedRange;
+
+  /**
+   * {@inheritDoc}
+   *
+   * <p>This implementation treats the claim's {@link ResourceTypeV2#createPhaseVersionString} value
+   * as a semantic version number and ensures that the number falls within the range allowed by this
+   * filter.
+   *
+   * @param resourceType used to interrogate the entity object
+   * @param entity the entity object
+   * @return true if the claim should be returned to clients
+   */
+  @Override
+  public boolean shouldRetain(ResourceTypeV2<?, ?> resourceType, Object entity) {
+    final var phaseVersionString = resourceType.createPhaseVersionString(entity);
```

Comment:
I can see why the `ClaimTypeV2` subclasses were used here, it seems convenient, though Claim/ClaimResponse doesn't play a part in that logic, which is a big part of those interface subclasses.  Wondering if a common interface on the entity would be simpler, then do like `<E extends PhasedEntity>`.

I think this would also have a small side benefit of being able to ditch all the wildcard type declarations we've previously had to use for `ClaimTypeV2` in the provider, etc.

Reply:
Thanks for the suggestion.  I was following the original pattern but I agree now might be a good time to revisit that.  I'll take a look.

Reply:
I have simplified the implementation by using an interface to recognize entities to filter.  It removed the need to modify the resource type stuff.  Can't do the generic type suggestion (`<E extends PhasedEntity>`) though because of the wildcard being used in `createBundleFor()`.  I like this implementation though since the filters can be applied to anything and just filter what they can and pass through anything else.

Reply:
OIC - disregard the backfill part of my reply.